stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/view/FileAttachmentsView.kt

```diff
@@ -121,20 +139,44 @@ private class FileAttachmentViewHolder(
         binding.apply {
             fileTypeIcon.loadAttachmentThumb(attachment)
             fileTitle.text = attachment.getDisplayableName()
-            fileSize.text = MediaStringUtil.convertFileSizeByteCount(attachment.fileSize.toLong())
 
-            if (attachment.uploadComplete == true || attachment.uploadComplete == null) {
-                actionButton.setImageResource(R.drawable.stream_ui_ic_icon_download)
-                val tintColor = ContextCompat.getColor(context, R.color.stream_ui_black)
-                ImageViewCompat.setImageTintList(actionButton, ColorStateList.valueOf(tintColor))
-            } else if (attachment.uploadComplete == false) {
+            if (attachment.uploadState is Attachment.UploadState.Failed) {
                 actionButton.setImageResource(R.drawable.stream_ui_ic_warning)
                 val tintColor = ContextCompat.getColor(context, R.color.stream_ui_accent_red)
                 ImageViewCompat.setImageTintList(actionButton, ColorStateList.valueOf(tintColor))
+                fileSize.text = MediaStringUtil.convertFileSizeByteCount(attachment.upload?.length() ?: 0L)
+            } else {
+                actionButton.setImageResource(R.drawable.stream_ui_ic_icon_download)
+                val tintColor = ContextCompat.getColor(context, R.color.stream_ui_black)
+                ImageViewCompat.setImageTintList(actionButton, ColorStateList.valueOf(tintColor))
+                fileSize.text = MediaStringUtil.convertFileSizeByteCount(attachment.fileSize.toLong())
+            }
+
+            attachment.uploadId?.let(ProgressTrackerFactory::getOrCreate)?.let { tracker ->
+                scope = CoroutineScope(DispatcherProvider.Main)
```

Comment:
Before creating the new scope, it's worth clearing the old one here, _just to be safe_